packages/client/hmi-client/src/components/workflow/tera-workflow.vue

```diff
@@ -465,7 +474,18 @@ function onMenuSelection(operatorType: string, menuNode: WorkflowNode<any>) {
 	newNodePosition.y = menuNode.y;
 
 	if (name && operation && node && drilldown) {
-		addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const newNode: WorkflowNode<any> = addOperatorToWorkflow({ name, operation, node, drilldown })();
+		const inputPort = newNode.inputs.find((input) => input.type.includes(port.type));
+
+		if (inputPort) {
+			// wait for the DOM to load the new node before adding the edge
+			await nextTick();
```

Comment:
Now that you're adding the edge differently, I wonder if that `nextTick()` is required anymore

Reply:
looks like it's still needed :(